src/mz/src/utils.rs

```diff
@@ -13,9 +13,66 @@ use indicatif::{ProgressBar, ProgressStyle};
 
 use crate::ExitMessage;
 
-/**
- * Trim lines. Useful when reading input data.
- */
+use clap::ArgEnum;
+
+/// Cloud providers and regions available.
+#[derive(Debug, Clone, ArgEnum)]
+pub(crate) enum CloudProviderRegion {
+    AwsUsEast1,
+    AwsEuWest1,
+}
+
+/// Implementation to name the possible values and parse every option.
+impl CloudProviderRegion {
+    pub fn variants() -> [&'static str; 2] {
+        ["aws/us-east-1", "aws/eu-west-1"]
+    }
+    pub fn parse(region: String) -> CloudProviderRegion {
+        match region.as_str() {
+            "aws/us-east-1" => CloudProviderRegion::AwsUsEast1,
+            "aws/eu-west-1" => CloudProviderRegion::AwsEuWest1,
+            _ => panic!("Unknown region."),
+        }
+    }
+    pub fn parse_region(region: String) -> &'static str {
+        match region.as_str() {
+            "aws/us-east-1" => "us-east-1",
+            "aws/eu-west-1" => "eu-west-1",
+            _ => panic!("Unknown region."),
+        }
+    }
+    pub fn parse_enum_region(region: CloudProviderRegion) -> &'static str {
+        match region {
+            CloudProviderRegion::AwsUsEast1 => "us-east-1",
+            CloudProviderRegion::AwsEuWest1 => "eu-west-1",
+        }
+    }
```

Comment:
These methods seem a bit clunky to use. Try this instead?

```suggestion
    /// Return the region name inside a cloud provider.
    pub fn region_name(&self) -> &'static str {
        match self {
            CloudProviderRegion::AwsUsEast1 => "us-east-1",
            CloudProviderRegion::AwsUsEast1 => "eu-west-1",
        }
    }
    
    /// Return the cloud provider name.
    pub fn cloud_provider(self) -> &'static str {
        match self {
            CloudProviderRegion::AwsUsEast1 |  CloudProviderRegion::AwsEuWest1 => "aws",
        }
    }
```